examples/prostate/prostate_2D/job_configs/prostate_fedavg/app/custom/learners/supervised_learner.py

```diff
@@ -0,0 +1,334 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
```

Comment:
If this code is the same as prostate_central/prostate_fedprox, 
Can you put them in a common directory and write a script to generate diff. jobs by copy them?

Similar as what we did in XGboost, what do you think?

@ZiyueXu77 @chesterxgchen

Reply:
Good point! what we did previously is to use separate "configs", while the customized python scripts were shared. Under job setting, I guess code need to be inside "app" (correct me if I am wrong), we can defenitely generate the jobs by copying them.

Reply:
done, I moved the custom part out, but kept the four config folders, as they can be fairly different due to different algorithms used, instead of just minor parameter ablation.